catalog/transformer/catalog-transformer-zip/src/main/java/org/codice/ddf/catalog/transformer/zip/ZipValidator.java

```diff
@@ -0,0 +1,151 @@
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.Enumeration;
+import java.util.Vector;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+import org.apache.wss4j.common.crypto.Merlin;
+import org.apache.wss4j.common.ext.WSSecurityException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.security.PropertiesLoader;
+
+public class ZipValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ZipValidator.class);
+
+    private String signaturePropertiesPath;
+
+    private static Merlin merlin;
+
+    public void init() {
+        try {
+            merlin = new Merlin(PropertiesLoader.loadProperties(signaturePropertiesPath),
+                    ZipDecompression.class.getClassLoader(),
+                    null);
+        } catch (WSSecurityException | IOException e) {
+            LOGGER.error("Unable to read merlin properties file.", e);
+        }
+    }
+
+    /**
+     * Validates a Zip file with the specified path.  If the Zip file does not have a manifest,
+     * is not signed, or does not match a certificate in the DDF KeyStore, the validation fails.
+     *
+     * @param filePath - the path to the zip file to validate
+     * @return true when the zip file is valid (signed by a trusted entity).
+     */
+    public boolean validateZipFile(String filePath) throws ZipValidationException{
+
+        try (JarFile jarFile = new JarFile(filePath)) {
+
+            Manifest man = jarFile.getManifest();
+
+            if (man == null) {
+                throw new ZipValidationException("Zip validation failed, missing manifest file.");
+            }
+
+            Vector entriesVec = new Vector<>();
+
+            byte[] buffer = new byte[ZipDecompression.BUFFER_SIZE];
+
+            Enumeration entries = jarFile.entries();
+
+            while (entries.hasMoreElements()) {
+                JarEntry je = (JarEntry) entries.nextElement();
+                if (je.isDirectory()) {
+                    continue;
+                }
+
+                entriesVec.addElement(je);
+                try (InputStream is = jarFile.getInputStream(je)) {
+
+                    while (is.read(buffer, 0, buffer.length) != -1) {
+                        // Read JarFile
+                    }
+                    is.close();
+                } catch (IOException e) {
+                    throw new ZipValidationException(String.format("Zip validation failed, unable to get input stream for entry %s", je.getName()));
+                }
+            }
+
+            Enumeration e = entriesVec.elements();
+            while (e.hasMoreElements()) {
+                JarEntry je = (JarEntry) e.nextElement();
+                Certificate[] certs = je.getCertificates();
+
+                if ((certs == null) || (certs.length == 0)) {
+                    if (!je.getName()
+                            .startsWith("META-INF")) {
```

Comment:
Will other steps in the ingest process be skipping the files in `META-INF`, or could something unsigned slip through by not being specific enough here?